expr/expr_test.go

```diff
@@ -2081,6 +2081,35 @@ func TestEvalExpression(t *testing.T) {
 			[]*MetricData{makeResponse("removeAbovePercentile(metric1, 50)",
 				[]float64{1, 2, -1, 7, math.NaN(), math.NaN(), math.NaN(), math.NaN()}, 1, now32)},
 		},
+		{
+			&expr{
+				target: "cactiStyle",
+				etype:  etFunc,
+				args: []*expr{
+					{target: "metric1"},
+					{valStr: "si", etype: etString},
+				},
+			},
+			map[MetricRequest][]*MetricData{
+				MetricRequest{"metric1", 0, 1}: {makeResponse("metric1", []float64{1, 2, 3, 4, 5, math.NaN()}, 1, now32)},
+			},
+			[]*MetricData{makeResponse("metric1 Current: 5 Max: 5 Min: 1", []float64{1, 2, 3, 4, 5, math.NaN()}, 1, now32)},
+		},
+		{
+			&expr{
+				target: "cactiStyle",
+				etype:  etFunc,
+				args: []*expr{
+					{target: "metric1"},
+					{valStr: "si", etype: etString},
+					{valStr: "carrots", etype: etString},
+				},
+			},
+			map[MetricRequest][]*MetricData{
+				MetricRequest{"metric1", 0, 1}: {makeResponse("metric1", []float64{1, 2, 3, 4, 5}, 1, now32)},
+			},
+			[]*MetricData{makeResponse("metric1 Current: 5 carrots Max: 5 carrots Min: 1 carrots", []float64{1, 2, 3, 4, 5}, 1, now32)},
+		},
```

Comment:
Added some of these examples + some random existing known data samples so the output is comparable to the Python impl.

Not sure how many tests this should have, as it seems to be quite verbose in the current setup. Perhaps could re-structure the tests into a loop or test multiple 'metrics' in 1 array entry or maybe just a smaller but more diverse data selection?